rockcraft.yaml

```diff
@@ -119,6 +121,14 @@ parts:
           echo "DOWNLOAD ERROR: spark-metrics-assembly-3.4-1.0.0.jar could not be downloaded properly! Exiting...." >&2
           exit 1
       fi
+      wget -q "https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-spark-runtime-${ICEBERG_SPARK_RUNTIME_VERSION}/${ICEBERG_VERSION}/iceberg-spark-runtime-${ICEBERG_SPARK_RUNTIME_VERSION}-${ICEBERG_VERSION}.jar"
+      wget -q "https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-spark-runtime-${ICEBERG_SPARK_RUNTIME_VERSION}/${ICEBERG_VERSION}/iceberg-spark-runtime-${ICEBERG_SPARK_RUNTIME_VERSION}-${ICEBERG_VERSION}.jar.sha1"
```

Comment:
**minor** Uhm, I believe we should pin the hash in the code. Otherwise if there is a malicious agent that can patch the JAR, he can also patch the SHA, and this safety check is rather useless. I understand that this would apply to the other JARS, but maybe we could do this for the others JARs as well, it should be pretty easy and straightforward. 

What do you think? @welpaolo @theoctober19th 

We could also factor that in a different PR, if you see that it goes out of the perimeter for this PR. I'm fine either way

Reply:
BTW...[here](https://github.com/canonical/charmed-kafka-snap/pull/34/files#diff-56759910381a014fecfd7556dd72ddd68c747d922a5b7df2044b9ce7c552f5f5R269) there is a similar code that includes jars using pinned sha:s. It is also somewhat easily extensible, since each jar is just a new line in the list. Maybe we could do this as well on spark (on different PR of course), where we just slightly refactor this? Seems more elegant, no?

Reply:
I think we need to pin SHAs in the rockcraft yaml for security reasons. I totally agree with that. I think that we need to adapt the solution you linked because we have multiple providers (maven.org, github).

Reply:
Updatd the PR to keep hardcoded SHA sums in variables and check them instead of downloading SHA sums on-the-fly.